src/hotspot/share/utilities/vmError.cpp

```diff
@@ -969,14 +988,15 @@ void VMError::report(outputStream* st, bool _verbose) {
     st->cr();
 
   STEP_IF("printing native stack (with source info)", _verbose)
-    address lastpc = nullptr;
     if (os::platform_print_native_stack(st, _context, buf, sizeof(buf), lastpc)) {
       // We have printed the native stack in platform-specific code
       // Windows/x64 needs special handling.
-      // Stack walking may got stuck. Try to print the calling code.
+      // Stack walking may get stuck. Try to find the calling code.
       if (lastpc != nullptr) {
-        st->print_cr("called by the following code:");
-        print_code(st, _thread, lastpc, true, printed, printed_capacity);
+        const char* name = find_code_name(lastpc);
+        if (name != nullptr) {
+          st->print_cr("the last pc belongs to %s (will be printed below)", name);
```

Comment:
s/the/The
s/will be//

Reply:
Why not print the nmethod right here?

Reply:
See my earlier comments. I was looking for some kind of delineation of all this additional stuff so that it would be easier to see where it fits into things. Not sure we've really achieved that regardless.